database/migration/migrateData.ts

```diff
@@ -380,6 +380,18 @@ const migrateData = async () => {
     console.log('Done migrating on v0.2.0...')
   }
 
+  // v0.3.1
+  if (databaseVersionLessThan('0.3.1')) {
+    console.log('Migrating to v0.3.1...')
+
+    // Enforce code must be unique per template in template_action table
+    console.log(' - Creating unique index for code/template in template_action')
+    await DB.changeSchema(`
+      CREATE UNIQUE INDEX IF NOT EXISTS unique_template_action_code
+      ON template_action (code, template_id)
```

Comment:
Just something I am thinking about, it's probably good idea to save version at the end of each migration (in case migration fails, we at least save version up to that last migration)

Reply:
You mean the version of the app, or the version of where we migrated to?

We currently save the database version in the "system_info" table, and that's what we compare to the current app version (from package.json) on start-up. Is that what you mean?

Reply:
the version that migration block is for:
```
if (databaseVersionLessThan('0.3.1')) {
    console.log('Migrating to v0.3.1...')

    // Enforce code must be unique per template in template_action table
    console.log(' - Creating unique index for code/template in template_action')
    await DB.changeSchema(`
      CREATE UNIQUE INDEX IF NOT EXISTS unique_template_action_code
      ON template_action (code, template_id)
   `)

    // Here -> 
    await DB.setDatabaseVersion('0.3.1')
  }
```

Reply:
Just food for thought btw, not required change